### ISSUE TITLE
False positive with \Ds\Vector::toArray()

### ISSUE THREAD

**Issue by enumag** [vimeo/psalm#3420]
```php
<?php declare(strict_types = 1);

namespace App;

use DateTimeImmutable;
use Ds\Vector;

abstract class Z
{
	/**
	 * @var Vector<DateTimeImmutable>
	 */
	private Vector $vector;

	public function test(): void
	{
		$this->process($this->vector->toArray());
	}

	/**
	 * @param iterable<DateTimeImmutable> $dates
	 */
	abstract function process(iterable $dates): void;
}
```

Based on [this line](https://github.com/vimeo/psalm/blob/952216ff21bd49ab930458c0cdf4c9cd642d75ed/src/Psalm/Internal/Stubs/ext-ds.php#L25), Psalm should be able to understand that `$this->vector->toArray()` returns `list<DateTimeImmutable>`. Unfortunately it doesn't understand that and reports this error:

```
ERROR: PossiblyInvalidArgument - src/Z.php:17:18 - Argument 1 of App\Z::process expects iterable<mixed, DateTimeImmutable>, possibly different type array<array-key, mixed> provided (see https://psalm.dev/092)
                $this->process($this->vector->toArray());
```

**Comment by muglug** [comment]
The problem here is that there was a separately-implemented method on the `Ds\Vector` extension class discovered via reflection, and Psalm doesn't override that return type with an inherited one from the `Ds\Collection` stub. I don't think that's a bug tbh.
